### ISSUE TITLE
Audit all receivers to make sure the follow the correct order operations

### ISSUE THREAD

**Issue by tigrannajaryan** [open-telemetry/opentelemetry-collector-contrib#5909]
See https://github.com/open-telemetry/opentelemetry-collector/pull/4262

**Comment by tigrannajaryan** [comment]
@open-telemetry/collector-maintainers is there a way to prevent certain issues from being closed as inactive? The fact that this was inactive for 120 days does not make it less true :-)

**Comment by atoulme** [comment]
You can add the "never stale" tag to the issue. I have done this now.

**Comment by atoulme** [comment]
Is this a manual audit? We can add a test as part of the component test suite but not sure how to test receivers that actively scrape endpoints.

**Comment by tigrannajaryan** [comment]
Automated tests are of course much more preferable. For periodic scrapers this is probably less important (if datapoints is lost typically nothing catastrophic happens since they are usually cumulative). I would focus on the network receivers.

**Comment by atoulme** [comment]
OK so if I was to be methodical about it, I would do this:
* [ ] add metadata entry that identifies which receivers do checkpointing (probably a warning entry)
* [ ] open a master issue for all those and for each, run an issue that manually audits the behavior
* [ ] open a separate issue down the line to investigate automated testing going forward.

**Comment by dmitryax** [comment]
@atoulme looks like a good plan, thanks! Couple Qs:

> add metadata entry that identifies which receivers do checkpointing (probably a warning entry)

Not sure I understand why a warning is needed here. Checkpointing is needed for any receiver doing continuous reading from an external source, e.g., a file. So it's a preferred behavior, not a warning.

> open a master issue for all those and for each, run an issue that manually audits the behavior

I believe this issue can serve that purpose. We can create subtasks here for each receiver.

**Comment by atoulme** [comment]
I agree on the warning thing. The status metadata currently can contain warnings, or at least that's what we're working towards. Warnings are over broad in their definition right now - you can think of them as "features", or "particularities'. One such warning being discussed is a component being "stateful" (with quotes because being stateful is itself a discussion). So if you prefer a more precise term, we can take that up and discuss. I just want to point out that we have a way to identify components that use checkpointing as part of their metadata.

**Comment by atoulme** [comment]
Peeling off #21049 as a separate issue to address this, lgtm!
